### ISSUE TITLE
Include :: multi level include may produce invalid SQL

### ISSUE THREAD

**Issue by maumar** [aspnet/EntityFramework#1362]
Consider the following model:

```
    public class Customer
    {
        public int Id { get; set; }
        public string Name { get; set; }

        public Order Order { get; set; }
    }

    public class Order
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public List<OrderDetail> OrderDetails { get; set; }
    }

    public class OrderDetail
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public Order Order { get; set; }
    }

    public class MyContext : DbContext
    {
        public DbSet<Customer> Customers { get; set; }
        public DbSet<Order> Orders { get; set; }
        public DbSet<OrderDetail> OrderDetails { get; set; }

        protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            modelBuilder.Entity<Customer>().OneToOne(c => c.Order);
            modelBuilder.Entity<Order>().OneToMany(o => o.OrderDetails);
            modelBuilder.Entity<Customer>().Property(o => o.Id).GenerateValueOnAdd(false);
            modelBuilder.Entity<Order>().Property(o => o.Id).GenerateValueOnAdd(false);
            modelBuilder.Entity<OrderDetail>().Property(o => o.Id).GenerateValueOnAdd(false);
        }

        protected override void OnConfiguring(DbContextOptions options)
        {
            options.UseSqlServer(new SqlConnectionStringBuilder { DataSource = ".", InitialCatalog = "MultiLevelInclude2", MultipleActiveResultSets = true, IntegratedSecurity = true }.ToString());
        }
    }

    class Program
    {
        static void Main(string[] args)
        {
            using (var ctx = new MyContext())
            {
                ctx.Database.EnsureDeleted();
                ctx.Database.EnsureCreated();


                var c1 = new Customer
                {
                    Id = 1,
                    Name = "Customer1"
                };

                var o1 = new Order
                {
                    Id = 1,
                    Name = "Order1"
                };

                var od10 = new OrderDetail
                {
                    Id = 10,
                    Name = "OrderDetail 1 0"
                };

                ctx.Customers.Add(c1);
                ctx.Orders.Add(o1);
                c1.Order = o1;

                ctx.SaveChanges();
            }

            using (var ctx = new MyContext())
            {
                var result = ctx.Customers.Include(c => c.Order.OrderDetails).ToList();
                Console.WriteLine(result);
            }
        }
    }
```

This will produce the following query:

```
SELECT [o].[Id], [o].[Name], [o].[OrderId]
FROM [OrderDetail] AS [o]
INNER JOIN (
    SELECT DISTINCT [c].[Id], [o].[Id]
    FROM [Customer] AS [c]
    LEFT JOIN [Order] AS [o] ON [o].[Id] = [c].[Id]
) AS [c] ON [o].[OrderId] = [c].[Id]
ORDER BY [c].[Id]```

This will throw the following exception:

Unhandled Exception: System.Data.SqlClient.SqlException: The column 'Id' was specified multiple times for 'c'.
   at System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.SqlInternalConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler,
TdsParserStateObject stateObj, Boolean& dataReady)
   at System.Data.SqlClient.SqlDataReader.TryConsumeMetaData()
   at System.Data.SqlClient.SqlDataReader.get_MetaData()
   at System.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, String resetOptionsString)
   at System.Data.SqlClient.SqlCommand.RunExecuteReaderTds(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, Boolean async, Int32 timeout, Task& task, Boolean asyncWrite, SqlDataReader ds)
   at System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method, TaskCompletionSource`1 completion, Int32 timeout, Task& task, Boolean asyncWrite)
   at System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method)
   at System.Data.SqlClient.SqlCommand.ExecuteReader(CommandBehavior behavior, String method)
   at System.Data.SqlClient.SqlCommand.ExecuteDbDataReader(CommandBehavior behavior)
   at System.Data.Common.DbCommand.ExecuteReader()
   at Microsoft.Data.Entity.Relational.Query.QueryingEnumerable`1.Enumerator.MoveNext() in D:\k\EntityFramework\src\EntityFramework.Relational\Query\QueryingEnumerable.cs:line 72
   at Microsoft.Data.Entity.Relational.Query.IncludeCollectionIterator.<GetRelatedValues>d__0.MoveNext() in D:\k\EntityFramework\src\EntityFramework.Relational\
Query\IncludeCollectionIterator.cs:line 36
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at Microsoft.Data.Entity.Query.QueryBuffer.Include(Object entity, IReadOnlyList`1 navigationPath, IReadOnlyList`1 relatedValueReaders, Int32 currentNavigationIndex) in D:\k\EntityFramework\src\EntityFramework.Core\Query\QueryBuffer.cs:line 184
   at Microsoft.Data.Entity.Query.QueryBuffer.Include(Object entity, IReadOnlyList`1 navigationPath, IReadOnlyList`1 relatedValueReaders) in D:\k\EntityFramework\src\EntityFramework.Core\Query\QueryBuffer.cs:line 154
   at Microsoft.Data.Entity.Relational.Query.QueryMethodProvider.<_IncludeCollection>d__0`1.MoveNext() in D:\k\EntityFramework\src\EntityFramework.Relational\Qu
ery\QueryMethodProvider.cs:line 58
   at Microsoft.Data.Entity.Query.LinqOperatorProvider.<_TrackEntities>d__0`2.MoveNext() in D:\k\EntityFramework\src\EntityFramework.Core\Query\LinqOperatorProvider.cs:line 29
   at Microsoft.Data.Entity.Query.EntityQueryExecutor.EnumerableExceptionInterceptor`1.EnumeratorExceptionInterceptor.MoveNext() in D:\k\EntityFramework\src\EntityFramework.Core\Query\EntityQueryExecutor.cs:line 185
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at BashMultiLevelInclude.Program.Main(String[] args) in D:\Projects\BashMultiLevelInclude\BashMultiLevelInclude\Program.cs:line 384

**Comment by maumar** [comment]
Problem is here:

SELECT [g].[CityOrBirthName], [g].[FullName], [g].[LeaderNickname], [g].[LeaderSquadId], [g].[Nickname], [g].[Rank], [g].[SquadId]
FROM [Gear] AS [g]
INNER JOIN (
    SELECT DISTINCT **[w].[Id], [s].[Id]**
    FROM [Weapon] AS [w]
    LEFT JOIN [Gear] AS [g] ON ([w].[OwnerNickname] = [g].[Nickname] AND [w].[OwnerSquadId] = [g].[SquadId])
    INNER JOIN [Squad] AS [s] ON [g].[SquadId] = [s].[Id]
) AS [w] ON [g].[SquadId] = [w].[Id]
ORDER BY [w].[Id]

**Comment by maumar** [comment]
fixed in f8b9dacab07c66e182b3f90e741f967dfd1377a2
